### PR TITLE
Ensure that `Projection`s and `ProcessManager`s declare at least one message handler during repository registration

### PR DESCRIPTION
**Pull request by armiol**
Before this PR it was possible to register `ProjectionRepository` and `ProcessManagerRepository` instances  in `BoundedContext`, even if the managed entity did not declare any handler methods (i.e. nothing marked with `@Subscribe` or `@React`).

At the same time such a check was already implemented for `AggregateRepository`.

This changeset addresses this behavior inconsistency and prevents Spine users from such a mistake.

Now, mandatory checks for the proper entity declaration are implemented. They are performed upon a repository registration in the `BoundedContext`. If an entity has no message handler methods, an `IllegalStateException` is raised — similar to the behavior implemented for `AggregateRepository`.

The version of the framework has been set to `0.9.77-SNAPSHOT`.

### CODE REVIEW

**Comment by armiol** [comment]
@alexander-yevsyukov PTAL.

**Comment by armiol** [comment]
@dmdashenkov PTAL.

File: server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java [hunk 1]
```diff
@@ -100,36 +103,70 @@ protected ProcessManagerRepository() {
...
-                          .register(commandDispatcher);
+        final boolean handlesCommands = register(boundedContext.getCommandBus(),
+                                                 DelegatingCommandDispatcher.of(this));
+        final boolean handlesDomesticRejections = register(boundedContext.getRejectionBus(),
+                                                           rejDispatcher);
+        final boolean handlesExternalRejections = register(boundedContext.getIntegrationBus(),
+                                                           rejDispatcher.getExternalDispatcher());
+        final boolean handlesDomesticEvents = !getMessageClasses().isEmpty();
+        final boolean handlesExternalEvents = !getExternalEventDispatcher().getMessageClasses()
+                                                                          .isEmpty();
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java]: Please fix the alignment.

File: server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java [hunk 2]
```diff
@@ -100,36 +103,70 @@ protected ProcessManagerRepository() {
...
+
+        final boolean subscribesToEvents = handlesDomesticEvents || handlesExternalEvents;
+        final boolean reactsUponRejections = handlesDomesticRejections || handlesExternalRejections;
+
+        if (!handlesCommands && !subscribesToEvents && !reactsUponRejections) {
+            throw newIllegalStateException(
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java]: This branch appears not covered.

File: server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java [hunk 3]
```diff
@@ -100,36 +103,70 @@ protected ProcessManagerRepository() {
...
-                          .register(extRejectionDispatcher);
+    /**
+     * Registers the given dispatcher in the bus if there is at least one message class declared
+     * for dispatching by the dispatcher.
+     *
+     * @param bus        the bus to register dispatchers in
+     * @param dispatcher the dispatcher to register
+     * @param <D>        the type of dispatcher
+     * @return
+     * {@code true} if there are message classes to dispatch by the given dispatchers,
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java]: This alignment appears strange.

File: server/src/main/java/io/spine/server/projection/ProjectionRepository.java [hunk 4]
```diff
@@ -91,6 +92,28 @@ EventStore getEventStore() {
...
+    @Override
+    public void onRegistered() {
+        super.onRegistered();
+
+        final boolean noEventSubscriptions = getMessageClasses().isEmpty();
+        if (noEventSubscriptions) {
+            final boolean noExternalSubscriptions = getExternalEventDispatcher().getMessageClasses()
+                                                                                .isEmpty();
+            if (noExternalSubscriptions) {
+                throw newIllegalStateException(
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/projection/ProjectionRepository.java]: This branch appears not covered.

File: server/src/main/java/io/spine/server/projection/ProjectionRepository.java [hunk 5]
```diff
@@ -91,6 +92,28 @@ EventStore getEventStore() {
...
+                        "Projections of the repository %s have neither domestic nor external " +
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/projection/ProjectionRepository.java]: Let's have `Projections of type ...` instead `of repository ...`. It seems more natural.

File: server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java (hunk 2 above)
Comment (dmdashenkov) [reply server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java]: This might be a bug on __codecov__, but please check for sure.

Comment (armiol) [reply server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java]: Double-checked. 

Yes, that looks like an outdated **codecov** report.

File: server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java (hunk 3 above)
Comment (armiol) [reply server/src/main/java/io/spine/server/procman/ProcessManagerRepository.java]: Adjusted. My goal was to have `{@code true}` and `{@code false}` inter-aligned.

File: server/src/main/java/io/spine/server/projection/ProjectionRepository.java (hunk 4 above)
Comment (armiol) [reply server/src/main/java/io/spine/server/projection/ProjectionRepository.java]: As per discussion, it actually is.

File: server/src/main/java/io/spine/server/projection/ProjectionRepository.java (hunk 5 above)
Comment (armiol) [reply server/src/main/java/io/spine/server/projection/ProjectionRepository.java]: As discussed, this an emphasis of a **repository**, which validation mechanism failed to complete a process of _Repository-in-Bounded-Context Registration_. 

So to me it makes sense.

**Pull request merged**